### ISSUE TITLE
release task re-applies all project plugins?

### ISSUE THREAD

**Issue by mttkay** [researchgate/gradle-release#126]
I'm not sure I understand what I'm seeing. Essentially, I'm writing a plugin that applies other plugins, including the `gradle-release` plugin:

```
// in my plugin
void apply(Project project) {
   ...
    def rootProject = project.rootProject
    if (!rootProject.plugins.hasPlugin('net.researchgate.release')) {
      rootProject.apply plugin: 'net.researchgate.release'
     ...
   }
}
```
The problem is that the conditional should only pass once, but it doesn't. The `release` task appears to be recreating all the project graphs, which leads to it applying *itself* again because of the above code. This leads to the task getting stuck in a loop and it keeps asking me for the current release version.
At least I'm seeing build output during the `release` task that I should only see once, during the initial configuration phase so I reckon it must somehow reconfigure the whole project?

Am I missing something?

**Comment by danez** [comment]
I really don't know whats happening. maybe in your plugin you can print the plugins of the root project. I'm also not sure if you can apply plugins to the root project. Not sure how the exact lifecycle with root and sub-projects work.

**Comment by townsfolk** [comment]
If I had to hazard a guess, this might be due to the plugin using the `GradleBuild` task type.

**Comment by danez** [comment]
I don't think this is an issue with the release plugin, but more like how you are applying it. We successfully apply the release plugin in our own plugin here at researchgate, though we don't have projects with subprojects.

**Comment by danez** [comment]
This is probably fixed after #116 got merged. If you want you can test it.

```
buildscript {
    repositories {
        maven { url 'http://oss.jfrog.org/artifactory/oss-snapshot-local' }
    }
    dependencies {
        classpath 'net.researchgate:gradle-release:2.3.0-SNAPSHOT'
    }
}
```

**Comment by mttkay** [comment]
Sweet, I'll give it another go soon and report back.
